sbom-utility-scripts/scripts/merge-cachi2-sboms-script/merge_cachi2_sboms.py

```diff
@@ -30,26 +80,29 @@ def _is_cachi2_non_registry_dependency(component: dict) -> bool:
 
     Note that this function is only applicable for PyPI or NPM components.
     """
-    purl = component.get("purl", "")
+    purl = component.purl()
+    if not purl:
+        return False
 
-    return (purl.startswith("pkg:pypi") or purl.startswith("pkg:npm")) and (
-        "vcs_url=" in purl or "download_url=" in purl
-    )
+    qualifiers = purl.qualifiers or {}
+    return purl.type in ("pypi", "npm") and ("vcs_url" in qualifiers or "download_url" in qualifiers)
 
 
-def _unique_key_cachi2(component: dict) -> str:
+def _unique_key_cachi2(component: SBOMItem) -> str:
     """
     Create a unique key from Cachi2 reported components.
 
     This is done by taking a purl and removing any qualifiers and subpaths.
 
     See https://github.com/package-url/purl-spec/tree/master#purl for more info on purls.
     """
-    url = urlsplit(component["purl"])
-    return url.scheme + ":" + url.path
+    purl = component.purl()
+    if not purl:
+        raise ValueError(f"cachi2 component with no purl? name={component.name()}, version={component.version()}")
+    return purl._replace(qualifiers=None, subpath=None).to_string()
```

Comment:
Just FYI: NamedTuple methods don't follow the same naming scheme as most other python classes. Despite starting with an underscore, the `_replace` method is public API https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace

(PackageURL is a NamedTuple)